docs/CONTRIBUTING.md

```diff
@@ -1,66 +1,97 @@
+more people provide feedback as early as possible.
+
+Significant code changes submitted without an accompanying proposal might
+be rejected and not merged. Such significant submissions without a proposal are
+discouraged to avoid wasting time.
+
 
 After design discussions:
-- Fork a private repo
-- Make the changes in the private repo
+- Fork the netplugin repository to your own public repository
+- Make the changes in your repository in a new branch
 - Add unit and system test cases for your code
 - Make sure existing tests and newly added tests pass
-- Merge your changes with the latest in the master branch
+- Rebase your branch on top of the latest master branch
 - Re-run the unit and system tests
-- Submit a PR with the code changes
-- This would involve discussions and few adjustments may need to be made.
-It is encouraged to engage into discussions during the coding phase as well.
-- After `LGTM` from maintainers, re-run unit and system tests
-- One of the maintainers would merge your changes into the appropriate release candidate
+- Submit a pull request with the code changes
+- A discussion may take place on your pull request
+- Discussing while writing the code is also recommended
+- Requested changes should be made to the same branch on your fork
+- Changes should be force pushed to the same branch of the pull request
+- The unit and system tests need to be run again after the maintainers `LGTM` the change
+- One of the maintainers will merge the changes
 
 
 ### Discussing use cases and requesting new features
-Bring up your use cases by submitting an issue. Describe the use case that
-is not handled in the latest version. Issues that seek new features should
-be titled as `Feature Request: ...`. It is encouraged to include diagrams 
-(pics of hand drawn diagrams is just fine), or other details that best
-describes a use case.
+Submit an issue to discuss your use case. A description of the use case
+should be provided. The description should also explain why the existing
+features don't help with this use case.
+Feature requests should have a title which starts with `Feature request: ...`.
+We encourage the inclusion of diagrams (or pictures of drawings) and other
+details to provide a better description of the use case.
+
 
 ### Becoming a maintainer
-Of course, play with the code, know is inside out - and you will know if you
-are ripe to become a maintainer. If you think you are ready, drop an eamil
-to any of the maintainers.
+Play with the code and know it inside out. Once you think you are comfortable
+with the code and you think you are ready to become a maintainer, you can send
+an email to one of the maintainers.
+
+### Commit message format guidelines
+The commit message should have a short summary of no more than 50
+characters on the first line. The description should use verbs in the imperative
+(e.g. `netmaster: fix bug`, not `netmaster: fixed bug`).
```

Comment:
Wonder why is this so critical... instead should we ask folks to specify the one liner to be succinct description of the fix itself. For example 'netmaster: fixed issue #114", or "netmaster: fix for race condition described in #114"